ui/components/multichain/asset-picker-amount/asset-picker-amount.tsx

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  Box,
+  Icon,
+  IconName,
+  IconSize,
+  Label,
+  Text,
+} from '../../component-library';
+import {
+  AlignItems,
+  BackgroundColor,
+  BorderColor,
+  BorderRadius,
+  BorderStyle,
+  Display,
+  IconColor,
+  TextColor,
+} from '../../../helpers/constants/design-system';
+
+import { AssetType } from '../../../../shared/constants/transaction';
+import UserPreferencedCurrencyInput from '../../app/user-preferenced-currency-input/user-preferenced-currency-input.container';
+import UserPreferencedTokenInput from '../../app/user-preferenced-token-input/user-preferenced-token-input.component';
+import {
+  getCurrentDraftTransaction,
+  updateSendAmount,
+} from '../../../ducks/send';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import UserPreferencedCurrencyDisplay from '../../app/user-preferenced-currency-display';
+import { PRIMARY } from '../../../helpers/constants/common';
+import TokenBalance from '../../ui/token-balance';
+import MaxClearButton from './max-clear-button';
+import AssetPicker from './asset-picker/asset-picker';
+
+// A component that combines an asset picker with an input for the amount to send.
+// Work in progress.
+export const AssetPickerAmount = () => {
+  const dispatch = useDispatch();
+  const t = useI18nContext();
+  const { asset, amount } = useSelector(getCurrentDraftTransaction);
+
+  if (!asset) {
+    throw new Error('No asset is drafted for sending');
+  }
+
+  return (
+    <Box className="asset-picker-amount">
+      <Box display={Display.Flex}>
+        <Label>{t('amount')}</Label>
+        <MaxClearButton />
+      </Box>
+      <Box
+        display={Display.Flex}
+        alignItems={AlignItems.center}
+        backgroundColor={BackgroundColor.backgroundDefault}
+        paddingLeft={3}
+        paddingRight={3}
+        borderRadius={BorderRadius.LG}
+        borderColor={
+          amount.error ? BorderColor.errorDefault : BorderColor.primaryDefault
+        }
+        borderStyle={BorderStyle.solid}
+        borderWidth={2}
+      >
+        <AssetPicker asset={asset} />
+
+        {asset.type === AssetType.native ? (
+          <UserPreferencedCurrencyInput
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore: I'm not sure why the types don't find `onChange`
+            onChange={(newAmount: string) =>
+              dispatch(updateSendAmount(newAmount))
+            }
+            hexValue={amount.value}
+            className="asset-picker-amount__input"
+            swapIcon={(onClick: React.MouseEventHandler) => (
+              <button
```

Comment:
Sorry for just seeing this now!  I think we should use `<ButtonIcon>` here, which with `backgroundColor={BackgroundColor.Transparent}`.

Reply:
done in https://github.com/MetaMask/metamask-extension/pull/21517/commits/87bba3ef8dc1983418b53dc9112d42fadecb51da